core/src/main/java/org/apache/iceberg/rest/requests/UpdateNamespacePropertiesRequest.java

```
@@ -0,0 +1,129 @@
+
+/**
+ * A REST request to set and/or remove properties on a namespace.
+ */
+public class UpdateNamespacePropertiesRequest {
+
+  private List<String> removals;
+  private Map<String, String> updates;
+
+  public UpdateNamespacePropertiesRequest() {
+    // Required for Jackson deserialization.
+  }
+
+  private UpdateNamespacePropertiesRequest(List<String> removals, Map<String, String> updates) {
+    this.removals = removals;
+    this.updates = updates;
+    validate();
+  }
+
+  UpdateNamespacePropertiesRequest validate() {
+    Preconditions.checkArgument(removals != null || updates != null,
+        "Invalid value for update namespace properties request: at least one of updates and removals must be non-null");
+    return this;
+  }
+
+  public List<String> removals() {
+    return removals == null ? ImmutableList.of() : removals;
+  }
+
+  public Map<String, String> updates() {
+    return updates == null ? ImmutableMap.of() : updates;
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("removals", removals)
+        .add("updates", updates)
+        .toString();
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static class Builder {
+    private final ImmutableSet.Builder<String> removalsBuilder = ImmutableSet.builder();
+    private final ImmutableMap.Builder<String, String> updatesBuilder = ImmutableMap.builder();
+
+    private Builder() {
+    }
+
+    public Builder remove(String removal) {
+      Preconditions.checkNotNull(removal,
+          "Invalid property to remove: null");
+      removalsBuilder.add(removal);
+      return this;
+    }
+
+    public Builder removeAll(Collection<String> removals) {
+      Preconditions.checkNotNull(removals,
+          "Invalid list of properties to remove: null");
+      Preconditions.checkArgument(!removals.contains(null),
+          "Invalid value in list of properties to remove: null");
```

I don't think this needs to distinguish between the list case. If an individual removal is null, it should just throw "Invalid property to remove: null" like the `remove(String)` case.

---

Ok. I'll update all of the ones that reference things like `list of properties` and `collection of properties` to try to use the same string message throughout.

---

Updated.